pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -895,16 +904,27 @@ public void redeliverUnacknowledgedMessages() {
     }
 
     @Override
-    public void redeliverUnacknowledgedMessages(List<MessageIdImpl> messageIds) {
+    public void redeliverUnacknowledgedMessages(Set<MessageIdImpl> messageIds) {
```

Comment:
Also, by knowing that messages are in sequence, removing them should have aproximately the same performance.
The problem lies when remove has to iterate the hole queue until finding the element.